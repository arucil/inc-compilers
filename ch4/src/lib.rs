#![feature(box_syntax)]
pub mod pass;