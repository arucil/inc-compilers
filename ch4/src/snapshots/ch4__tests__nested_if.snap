---
source: ch4/src/lib.rs
expression: prog
---
extern rt_print_int, rt_print_newline, rt_read_int
section .text

    global _start
_start:
    push rbp
    mov rbp, rsp
    sub rsp, 8
    push rbx
    call rt_read_int
    mov rbx, rax
    call rt_read_int
    mov rcx, rax
    cmp rbx, 1
    jl .block2
    cmp rbx, 2
    je .block0

.block1:
    mov rax, rcx
    add rax, 10

.conclusion:
    mov rdi, rax
    call rt_print_int
    call rt_print_newline
    pop rbx
    mov rsp, rbp
    pop rbp
    mov rax, 60
    mov rdi, 0
    syscall

.block0:
    mov rax, rcx
    add rax, 2
    jmp .conclusion

.block2:
    cmp rbx, 0
    je .block0
    jmp .block1

